File: Mozu.SiteBuilder.UX.Models/Customers/Address.cs
using System.Runtime.Serialization;

namespace Mozu.SiteBuilder.UX.Models.Customers
{
    [DataContract]
    public class Address
    {
        [DataMember(Name = "addressId")]
        public int? AddressId { get; set; }

        [DataMember(Name = "address1")]
        public string Address1 { get; set; }

        [DataMember(Name = "address2")]
        public string Address2 { get; set; }

        [DataMember(Name = "address3")]
        public string Address3 { get; set; }

        [DataMember(Name = "cityOrTown")]
        public string CityOrTown { get; set; }

        [DataMember(Name = "stateOrProvince")]
        public string StateOrProvince { get; set; }

        [DataMember(Name = "postalOrZipCode")]
        public string PostalOrZipCode { get; set; }

        [DataMember(Name = "countryCode")]
        public string CountryCode { get; set; }
    }
}

File: Mozu.SiteBuilder.UX.Admin/Scripts/app/model/Address.js
/**
 * @class Taco.model.Address
 */
Ext.define('Taco.model.Address', {
    extend: 'Taco.core.data.Model',
    fields: [
        { name: 'firstName',             type: 'string' },
        { name: 'lastName',              type: 'string' },
        { name: 'middleNameOrInitial',   type: 'string' },
        { name: 'lastNameOrSurname',     type: 'string' },
        { name: 'companyOrOrganization', type: 'string' },
        { name: 'address1',              type: 'string' },
        { name: 'address2',              type: 'string' },
        { name: 'address3',              type: 'string' },
        { name: 'cityOrTown',            type: 'string' },
        { name: 'stateOrProvince',       type: 'string' },
        { name: 'countryCode',           type: 'string' },
        { name: 'postalOrZipCode',       type: 'string' },
        { name: 'addressIsValidated',    type: 'boolean' },
        { name: 'phoneNumber',           type: 'string' }
    ],

    proxy: {
        type: 'ajaxproxy',
        api: {
            create: '/admin/app/address/create',
            read: '/admin/app/address/read',
            update: '/admin/app/address/update',
            destroy: '/admin/app/address/delete',
            duplicate: '/admin/app/address/duplicate'
        },
        reader: {
            type: 'json',
            root: 'items',
            successProperty: 'success'
        },
        writer: {
            type: 'json'
        }
    }
});
